velox/functions/prestosql/benchmarks/ArrayMinMaxBenchmark.cpp

```diff
@@ -79,21 +136,58 @@ struct ArrayMaxSimpleFunction {
   VELOX_DEFINE_FUNCTION_TYPES(T);
 
   template <typename TInput>
-  FOLLY_ALWAYS_INLINE bool call(TInput& out, const arg_type<Array<TInput>>& x) {
-    if (x.begin() == x.end()) {
-      return false;
+  FOLLY_ALWAYS_INLINE bool call(
+      TInput& out,
+      const arg_type<Array<TInput>>& array) {
+    const auto size = array.size();
+    if (size == 0) {
+      return false; // NULL
     }
-    out = std::max_element(x.begin(), x.end())->value();
+
+    auto max = INT32_MIN;
+    if (array.mayHaveNulls()) {
+      for (const auto& item : array) {
+        if (!item.has_value()) {
+          return false; // NULL
+        }
+        if (item.value() > max) {
+          max = item.value();
+        }
+      }
+    }
+
+    for (const auto& item : array) {
```

Comment:
shouldn't there be an else here?

Reply:
i copy pasted the vector function implementation here and then disabled the fast path

Reply:
yes i disabled running those in the benchmarks since min is enough, i will just remove the function